File: server/src/main/java/com/github/genraven/genesys/model/Skill.java
package com.github.genraven.genesys.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "skill")
@Getter
@Setter
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "type")
    private Type type;

    public enum Type {
        GENERAL, MAGIC, SOCIAL, KNOWLEDGE, COMBAT;
    }
}

File: client/src/models/actor/Skill.ts
import {CharacteristicType} from '../character/Characteristic';

export default interface Skill {
    id: string
    characteristic: CharacteristicType
    type: SkillType
    name: string
    initiative: boolean
};

export enum SkillType {
    General = 'General',
    Magic = 'Magic',
    Social = 'Social',
    Combat = 'Combat',
    Knowledge = 'Knowledge'
}